From 94f75e80508c4fb7543bc0d59ce4e651d6b839b2 Mon Sep 17 00:00:00 2001
From: Huang Shijie <b32955@freescale.com>
Date: Mon, 28 Nov 2011 17:50:11 +0800
Subject: [PATCH 0680/2463] ENGR00163603-4 MX6Q SDMA firmware : fix a bug in
 the UART receiving.

The uart_to_mcu code in the ROM has bug.
The new fireware fixes it.

Signed-off-by: Huang Shijie <b32955@freescale.com>
---
 firmware/imx/sdma/sdma-imx6q-to1.bin.ihex |  197 ++++++++++++++---------------
 1 files changed, 95 insertions(+), 102 deletions(-)

diff --git a/firmware/imx/sdma/sdma-imx6q-to1.bin.ihex b/firmware/imx/sdma/sdma-imx6q-to1.bin.ihex
index d180a30..04ae992 100644
--- a/firmware/imx/sdma/sdma-imx6q-to1.bin.ihex
+++ b/firmware/imx/sdma/sdma-imx6q-to1.bin.ihex
@@ -1,106 +1,99 @@
-:1000000053444D4101000000000000001C000000AE
-:1000100023000000A8000000DC05000082020000B0
+:1000000053444D4101000000010000001C000000AD
+:1000100023000000A80000006A0500008202000022
 :10002000FFFFFFFF00000000FFFFFFFFFFFFFFFFDC
 :10003000FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFD0
-:10004000FFFFFFFFFFFFFFFF31030000FFFFFFFF88
-:10005000EB0200003F190000FFFFFFFF0804000053
-:10006000FFFFFFFFC0030000FFFFFFFFFFFFFFFFD9
-:10007000FFFFFFFFAB020000FFFFFFFF7B0300005D
+:10004000FFFFFFFFB61800006A1A00004B04000013
+:10005000EB0200004A190000FF180000080400002D
+:1000600095040000C0030000C105000070050000F9
+:1000700009040000AB020000E30400007B03000061
 :10008000FFFFFFFFFFFFFFFF4C0400006E040000B6
-:10009000FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF70
-:1000A0000000000000180000700D11051555CD5A14
-:1000B000FD5F010B6606017C17036706027C1703D0
-:1000C00017036806047C1703170317031703C55BA0
-:1000D0008D073CC1637D4EC1617C081DD75ADF5B33
-:1000E000E75CEF5DF75E8803700E1106FF0A1102F0
-:1000F000FF1ABC02800CD402017C8A04A4022E5593
-:10010000B905004D057CFE55BB05004D017C2B98C3
-:10011000FF55E7507800017C081D216DF755056DEE
-:10012000081DF75D7DD81F7F080818681C7C286DA6
-:100130001A7F0824027C004A1B7DFF55E750780097
-:10014000017D081D056DEF55016D081DEF5D080867
-:100150001868097C286D077F004CD17C0000269828
-:10016000DF5BE75C880307000C68E7545404E75C36
-:100170008B00D752DF53E75459C1107C4EC10E7C1F
-:10018000DF5BFF527803017D081A016D056A0808DC
-:100190001868E67C286AE47FD75219987DD8000059
-:1001A0001598CF55D26DC755C86DDE6D010206009A
-:1001B000E3C1DB57E35FE357F352016A8F00D500D9
-:1001C000017D8D00A005EB5D7804037D79042C7D15
-:1001D000367C79041F7CEE56000F6006057D0965AC
-:1001E000437E0A62417EA4980A623E7E09653C7E97
-:1001F00012051205AD026007037DFB55D36DAF9864
-:10020000FB55041DD36DC86A2F7F011F03200048D2
-:10021000E47CD798FB55D76D150005780962C86A4C
-:100220000962C86AD76DD698FB55D76D15001500C1
-:1002300005780A62C86A0A62C86AD76DD698FB5503
-:10024000D76D15001500150005780B62C86A0B62A2
-:10025000C86AD76D097CDF6D077F0000EB55004D44
-:10026000077DFAC1E3578A980700CC680C6813C26F
-:100270000AC28798D9C1E3C1DB57E35FE357F35262
-:10028000216A8F00D500017D8D00A005EB5DFB5636
-:100290007804037D79042A7D317C7904207C700BFD
-:1002A0001103EB53000F6003057D0965377E0A6279
-:1002B000357E0A990A62327E0965307E1205120582
-:1002C000AD026007027C065A1299265A277F011F49
-:1002D00003200048E87C700B110313533399150079
-:1002E00004780962065A0962265A329915001500E7
-:1002F00004780A62065A0A62265A329915001500D5
-:10030000150004780B62065A0B62265A077C00001F
-:10031000EB55004D067DFAC1E357ED9807000C68D8
-:1003200013C20AC2EA98700B110313536C07017CC5
-:10033000D9C1FB5E8A066B07017CD9C1F35EDB592C
-:10034000D3588F0110010F398B003CC12B7DC05A4F
-:10035000C85B4EC1277C88038906E35CFF0D11054D
-:10036000FF1DBC053E07004D187D700811007E077B
-:10037000097D7D07027D28526A99F852DB54BC0240
-:10038000CC02097C7C07027D28527399F852D35421
-:10039000BC02CC02097D000461998B00C052C85395
-:1003A00059C1D67D00025199FF08BF007F07157D16
-:1003B0008804D500017D8D00A005EB5D8F0212023F
-:1003C0001202FF3ADA05027C3E079C99A402DD0284
-:1003D000027D3E079C995E079C99EB559805EB5D65
-:1003E000F352FB546A07267D6C07017DD9996B0790
-:1003F000577C6907047D6807027D010EB399935805
-:10040000D600017D8E009355A005935DA00602786D
-:100410000255045D1D7C004E087C6907037D025572
-:10042000177EC099045D147F890693500048017DB2
-:10043000AB99249A150006780255045D4F070255C2
-:10044000245D2F07017C249A17006F07017C01208F
-:1004500093559D0007002BDA7999D36C6907047DC9
-:100460006807027D010EE8999358D600017D8E0041
-:100470009355A005935DA00602780255C86D0F7CC8
-:10048000004E087C6907037D0255097EF599C86D09
-:10049000067F890693500048017DE099249A1E9AB0
-:1004A000C36A6907047D6807027D010E0B9A9358A1
-:1004B000D600017D8E009355A005935DA0060278BD
-:1004C000C865045D0F7C004E087C6907037DC86524
-:1004D000097E189A045D067F890693500048017DC5
-:1004E000039A249A93559D000700FF6C2BDA7999A3
-:1004F0000000E354EB55004D017C79996199E35478
-:10050000EB55FF0A1102FF1A7F07027CA005389AFB
-:100510009D008C05BA05A0051002BA04AD04540470
-:100520000600E3C1DB57FB52C36AF352056A8F0032
-:10053000D500017D8D00A005EB5D7804037D790475
-:100540002B7D1E7C7904337CEE56000FFB55600733
-:10055000027DC36D599A041DC36DC8623B7E60065F
-:10056000027D10021202096A357F1202096A327F87
-:100570001202096A2F7F011F03200048E77C8D9A31
-:10058000FB55C76D1500150015000578C8620B6A8C
-:10059000C8620B6AC76D8C9AFB55C76D15001500B4
-:1005A0000578C8620A6AC8620A6AC76D8C9AFB55E8
-:1005B000C76D15000578C862096AC862096AC76D07
-:1005C000097C286A077F0000EB55004D057DFAC1C4
-:1005D000DB57439A77C254040AC23E9AD9C1E3C199
-:1005E000DB57F352056A8F00D500017D8D00A00511
-:1005F000FB567804037D7904297D1F7C79042E7CC9
-:10060000E35D700D1105ED55000F6007027D065288
-:10061000B69A2652337E6005027D10021202096AE4
-:100620002D7F1202096A2A7F1202096A277F011FA1
-:1006300003200048EA7CE355E19A150015001500F7
-:10064000047806520B6A26520B6AE09A15001500D0
-:10065000047806520A6A26520A6AE09A150004785B
-:100660000652096A2652096A097C286A077F000037
-:10067000DB57004D057DFAC1DB579F9A77C25404C2
-:040680000AC29C9A74
+:10009000FFFFFFFF00180000FFFFFFFFFFFFFFFF54
+:1000A0000000000000180000E3C1DB57E35FE357E6
+:1000B000F352016A8F00D500017D8D00A005EB5D34
+:1000C0007804037D79042C7D367C79041F7CEE5600
+:1000D000000F6006057D0965437E0A62417E209817
+:1000E0000A623E7E09653C7E12051205AD0260077C
+:1000F000037DFB55D36D2B98FB55041DD36DC86A4A
+:100100002F7F011F03200048E47C5398FB55D76DD7
+:10011000150005780962C86A0962C86AD76D5298E5
+:10012000FB55D76D1500150005780A62C86A0A628A
+:10013000C86AD76D5298FB55D76D1500150015008C
+:1001400005780B62C86A0B62C86AD76D097CDF6DDF
+:10015000077F0000EB55004D077DFAC1E357069875
+:100160000700CC680C6813C20AC20398D9C1E3C166
+:10017000DB57E35FE357F352216A8F00D500017D1F
+:100180008D00A005EB5DFB567804037D79042A7D84
+:10019000317C7904207C700B1103EB53000F60035A
+:1001A000057D0965377E0A62357E86980A62327E51
+:1001B0000965307E12051205AD026007027C065A01
+:1001C0008E98265A277F011F03200048E87C700B79
+:1001D00011031353AF98150004780962065A096297
+:1001E000265AAE981500150004780A62065A0A626B
+:1001F000265AAE9815001500150004780B62065AB1
+:100200000B62265A077C0000EB55004D067DFAC1B3
+:10021000E357699807000C6813C20AC26698700B0E
+:10022000110313536C07017CD9C1FB5E8A066B076F
+:10023000017CD9C1F35EDB59D3588F0110010F390E
+:100240008B003CC12B7DC05AC85B4EC1277C880304
+:100250008906E35CFF0D1105FF1DBC053E07004D3F
+:10026000187D700811007E07097D7D07027D2852E8
+:10027000E698F852DB54BC02CC02097C7C07027D74
+:100280002852EF98F852D354BC02CC02097D0004E6
+:10029000DD988B00C052C85359C1D67D0002CD985D
+:1002A000FF08BF007F07157D8804D500017D8D0004
+:1002B000A005EB5D8F0212021202FF3ADA05027C02
+:1002C0003E071899A402DD02027D3E0718995E07D9
+:1002D0001899EB559805EB5DF352FB546A07267DA0
+:1002E0006C07017D55996B07577C6907047D68078A
+:1002F000027D010E2F999358D600017D8E009355F3
+:10030000A005935DA00602780255045D1D7C004E99
+:10031000087C6907037D0255177E3C99045D147FB4
+:10032000890693500048017D2799A0991500067809
+:100330000255045D4F070255245D2F07017CA099EB
+:1003400017006F07017C012093559D000700A7D976
+:10035000F598D36C6907047D6807027D010E6499E6
+:100360009358D600017D8E009355A005935DA0069D
+:1003700002780255C86D0F7C004E087C6907037D2A
+:100380000255097E7199C86D067F89069350004811
+:10039000017D5C99A0999A99C36A6907047D6807F1
+:1003A000027D010E87999358D600017D8E009355EA
+:1003B000A005935DA0060278C865045D0F7C004E21
+:1003C000087C6907037DC865097E9499045D067FF2
+:1003D000890693500048017D7F99A09993559D000F
+:1003E0000700FF6CA7D9F5980000E354EB55004DCA
+:1003F000017CF598DD98E354EB55FF0A1102FF1AD2
+:100400007F07027CA005B4999D008C05BA05A00564
+:100410001002BA04AD0454040600E3C1DB57FB52DA
+:10042000C36AF352056A8F00D500017D8D00A005D7
+:10043000EB5D7804037D79042B7D1E7C7904337C8D
+:10044000EE56000FFB556007027DC36DD599041D64
+:10045000C36DC8623B7E6006027D10021202096A0B
+:10046000357F1202096A327F1202096A2F7F011F4B
+:1004700003200048E77C099AFB55C76D150015005D
+:1004800015000578C8620B6AC8620B6AC76D089AC6
+:10049000FB55C76D150015000578C8620A6AC86269
+:1004A0000A6AC76D089AFB55C76D15000578C862C2
+:1004B000096AC862096AC76D097C286A077F00005B
+:1004C000EB55004D057DFAC1DB57BF9977C2540447
+:1004D0000AC2BA99D9C1E3C1DB57F352056A8F004A
+:1004E000D500017D8D00A005FB567804037D7904BD
+:1004F000297D1F7C79042E7CE35D700D1105ED557F
+:10050000000F6007027D0652329A2652337E600544
+:10051000027D10021202096A2D7F1202096A2A7FE7
+:100520001202096A277F011F03200048EA7CE35575
+:100530005D9A150015001500047806520B6A2652C4
+:100540000B6A5C9A15001500047806520A6A265256
+:100550000A6A5C9A150004780652096A2652096AEA
+:10056000097C286A077F0000DB57004D057DFAC132
+:10057000DB571B9A77C254040AC2189AE3C1DB57AF
+:10058000F352056AFB568E02941AC36AC862690266
+:100590001D7D941EC36ED36EC8624802C86A94263D
+:1005A000C36E981EC36EC8629826C36E6002097C33
+:1005B000C8626E02247D096A1E7F0125004D257DDB
+:1005C0007D9A286A187F04627AC2B19AE36E8F001E
+:1005D000D805017D8D00A005C8626E02107D096AF4
+:1005E0000A7F0120F97C286A067F0000004D0D7DFE
+:1005F000FAC1DB576E9A070004620C6AAE9A286A49
+:10060000FA7F04627AC258045404286AF47F0AC24A
+:020610006B9AE3
 :00000001FF
-- 
1.7.7.4

